### ISSUE TITLE
Remove Unix config of Microsoft.Win32.Registry

### ISSUE THREAD

**Issue by danmoseley** [dotnet/runtime#46209]
We [ship a runtime-only copy on Unix ](https://github.com/dotnet/runtime/blob/75f83464af09e441a1a52cf4f6fe8251477273c9/src/libraries/NetCoreAppLibrary.props#L149) that has part of the code, but ultimately throws when you try to open a key.

It seems like this is only needed by Microsoft.VisualBasic.Interaction. Making a partial class for that for Unix that just throws directly would allow us to remove the Unix config of M.W32.Registry. It's only 60KB though.

This just popped out when using ILSPY to see what in a Linux published app has SupportedOSPlatformAttribute or UnsupportedOSPlatformAttribute applied.

**Comment by ericstj** [comment]
> Also, do we still need to include Microsoft.Win32.Registry.dll in Unix?

Yes, at least given how we build today.  Assumptions are made by the SDK that we have the same set of assemblies on all runtimes.  This lets the SDK know when it needs to include an assembly in a framework dependent app or not.  We wouldn't want a runtime agnostic app to depend on registry and get a FNFE instead of PNSE on Unix.  We could lie about registry being in the framework, but then apps would carry it even on Windows.  We could ask the SDK to have RID-specific file lists, but then we'd still be in a place where a RID-agnostic app carried the assembly (since it might run on Unix).  For this reason we make sure that all frameworks carry the same set of assemblies.

>  The Microsoft.Win32.Registry on non-Windows is wierd - it has much of the code, but will eventually throw PNSE when it ends up accessing the registry. My guess this is in anticipation of having a file system implementation, hence there's RegistryKey.FileSystem.cs, but it's just PNSE too.

I've wondered this too.  My assumption was that it was done for mono-compat.  Looks like @stephentoub did it explicitly here: https://github.com/dotnet/corefx/pull/9456

**Comment by danmosemsft** [comment]
Is my change linked above worthwhile? Seems cleanup.

**Comment by ericstj** [comment]
I don’t think it’s worthwhile to remove the dependency.  I do think it’s worthwhile to simplify registry to be a full PNSE assembly if we don’t need it to work.

**Comment by stephentoub** [comment]
We added that set of stubs when we thought that a) we would need it for mono compat and b) it would be a good project for an intern to implement.  That was several years ago, though.  If @marek-safar says it's not needed, I agree we should just delete the stubs.
